sycl/include/sycl/types.hpp

```diff
@@ -907,8 +903,23 @@ template <typename Type, int NumElements> class vec {
             typename = typename std::enable_if_t<
                 std::is_same<vector_t_, vector_t>::value &&
                 !std::is_same<vector_t_, DataT>::value>>
-  constexpr vec(vector_t openclVector) : m_Data(openclVector) {}
-  operator vector_t() const { return m_Data; }
+  constexpr vec(vector_t openclVector) {
+    if constexpr (!IsUsingArray) {
+      m_Data = openclVector;
+    } else {
+      detail::loop<NumElements>(
+          [&, this](size_t i) { m_Data[i] = openclVector[i]; });
+    }
+  }
+
+  operator vector_t() const {
+    if constexpr (!IsUsingArray) {
```

Comment:
So, we never convert to `ext_vector_size` when the size if bigger than 64? Right?